src/cljs/org/broadinstitute/firecloud_ui/page/workspace/submission_details.cljs

```diff
@@ -126,9 +126,41 @@
                    (:workflows props))}])))})
 
 
+(react/defc AbortButton
+  {:kill-submission
+   (fn [{:keys [props state this]}]
+     (swap! state assoc :stopping-submission? true)
+     (let [submission-id (get (:submission props) "submissionId" )
+           workspace-id (:workspace-id props)
+           path (paths/get-abort-submission-path workspace-id submission-id)
+           headers {"Content-Type" "application/json"}
+           canned-response {:responseText ""
+                            :status 200 :delay-ms (rand-int 2000)}
+           on-done (fn [{:keys [success? status-text]}]
+                     (swap! state assoc :stopping-submission? false)
+                     (if success?
+                       ((:on-kill props))
+                       (js/alert (str "Error in stopping the job : " status-text))))]
```

Comment:
Ok, so now the AbortButton only uses the submission's ID, so it can take that instead of the whole submission object.  I still think you're over-using `let`: We don't need to bind submission-id and workspace-id since we're just using them in once place.  Similarly, binding the path and headers doesn't add any value.  On line 137 you can just get rid of the `:responseText` key.